server/service.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	pb "github-search-service/api"
+)
+
+type githubSearchServer struct {
+	pb.UnimplementedGithubSearchServiceServer
+}
+
+type GitHubItem struct {
+	HTMLURL string `json:"html_url"`
+	Repo    struct {
+		FullName string `json:"full_name"`
+	} `json:"repository"`
+}
+
+type GitHubResponse struct {
+	Items []GitHubItem `json:"items"`
+}
+
+func (s *githubSearchServer) Search(ctx context.Context, req *pb.SearchRequest) (*pb.SearchResponse, error) {
```

Comment:
Should implement some sort of logging so that it is easy to see what the server is doing.